src/RoadRegistry.Syndication.Projections/MunicipalityCacheProjection.cs

```diff
@@ -0,0 +1,183 @@
+namespace RoadRegistry.Syndication.Projections
+{
+    using System;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Connector;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore;
+    using MunicipalityEvents;
+    using Schema;
+
+    public class MunicipalityCacheProjection : ConnectedProjection<SyndicationContext>
+    {
+        public MunicipalityCacheProjection()
+        {
+            When<Envelope<MunicipalityWasRegistered>>(async (context, envelope, token) =>
+            {
+                await context.Municipalities.AddAsync(
+                    new MunicipalityRecord
+                    {
+                        MunicipalityId = envelope.Message.MunicipalityId,
+                        NisCode = envelope.Message.NisCode,
+                        DutchName = null,
+                        FrenchName = null,
+                        GermanName = null,
+                        EnglishName = null,
+                        MunicipalityStatus = MunicipalityStatus.Registered
+                    }, token);
+            });
+
+            When<Envelope<MunicipalityWasNamed>>(async (context, envelope, token) =>
+            {
+                var municipalityRecord = await context.Municipalities.FindAsync(envelope.Message.MunicipalityId);
+                if (municipalityRecord == null)
+                    return;
+
+                switch (envelope.Message.Language)
+                {
+                    case MunicipalityLanguage.Dutch:
+                        municipalityRecord.DutchName = envelope.Message.Name;
+                        break;
+                    case MunicipalityLanguage.French:
+                        municipalityRecord.FrenchName = envelope.Message.Name;
+                        break;
+                    case MunicipalityLanguage.German:
+                        municipalityRecord.GermanName = envelope.Message.Name;
+                        break;
+                    case MunicipalityLanguage.English:
+                        municipalityRecord.EnglishName = envelope.Message.Name;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
```

Comment:
This is simply obscure and breaks for no interesting reason. The choice is to either throw a specific exception that indicates we do not support the passed in language OR to simply not handle this case at all. I'd be inclined to opt for the latter since we're already pretty deliberate about our supported languages. Next to that, given that MunicipalityLanguage is an enum, I doubt any other string would deserialize properly into it.